tests/test-plugins/activity-container/src/main.rs

```diff
@@ -0,0 +1,159 @@
+	weeks: Option<u16>,
+}
+
+static CONFIG: OnceLock<Config> = OnceLock::new();
+
+/// Returns the span of time since the most recent commit to a Git repo as `jiff:Span` displayed as a String
+/// (Which means that anything expecting a `Span` must parse the output of this query appropriately)
+#[query(default)]
+async fn activity(engine: &mut PluginEngine, target: Target) -> Result<String> {
+	log::debug!("running activity query");
+
+	let repo = target.local;
+
+	// Get today's date
+	let today = Timestamp::now();
+
+	// Get the date of the most recent commit.
+	let value = engine
+		.query("mitre/git/last_commit_date", repo)
+		.await
+		.map_err(|e| {
+			log::error!("failed to get last commit date for activity metric: {}", e);
+			Error::UnspecifiedQueryState
+		})?;
+
+	let Value::String(date_string) = value else {
+		return Err(Error::UnexpectedPluginQueryInputFormat);
+	};
+	let last_commit_date: Timestamp = date_string.parse().map_err(|e| {
+		log::error!("{}", e);
+		Error::UnspecifiedQueryState
+	})?;
+
+	// Get the time between the most recent commit and today.
+	let time_since_last_commit = today.since(last_commit_date).map_err(|e| {
+		log::error!("{}", e);
+		Error::UnspecifiedQueryState
+	})?;
+
+	Ok(time_since_last_commit.to_string())
+}
+
+#[derive(Clone, Debug)]
+struct ActivityPlugin;
+
+impl Plugin for ActivityPlugin {
+	const PUBLISHER: &'static str = "mitre";
+
+	const NAME: &'static str = "activity";
+
+	fn set_config(&self, config: Value) -> StdResult<(), ConfigError> {
+		let conf =
+			serde_json::from_value::<Config>(config).map_err(|e| ConfigError::Unspecified {
+				message: e.to_string(),
+			})?;
+		CONFIG.set(conf).map_err(|_e| ConfigError::Unspecified {
+			message: "config was already set".to_owned(),
+		})
+	}
+
+	fn default_policy_expr(&self) -> Result<String> {
+		let Some(conf) = CONFIG.get() else {
+			log::error!("tried to access config before set by Hipcheck core!");
+			return Err(Error::UnspecifiedQueryState);
+		};
+
+		Ok(format!("(lte $ P{}w)", conf.weeks.unwrap_or(71)))
+	}
+
+	fn explain_default_query(&self) -> Result<Option<String>> {
+		Ok(Some(
+			"span of time that has elapsed since last activity in repo".to_string(),
+		))
+	}
+
+	queries! {}
+}
+
+#[derive(Parser, Debug)]
+struct Args {
+	#[arg(long)]
+	port: u16,
+
+	#[arg(trailing_var_arg(true), allow_hyphen_values(true), hide = true)]
+	unknown_args: Vec<String>
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<()> {
+	let args = Args::try_parse().unwrap();
+	log::info!("Activity container plugin is registering {:?}", args);
+	PluginServer::register(ActivityPlugin {})
+		.listen(Host::Any, args.port)
+		.await
+}
+
+#[cfg(test)]
+mod test {
+	use super::*;
+
+	use hipcheck_sdk::types::LocalGitRepo;
+	use jiff::{Span, SpanRound, Unit};
+	use std::result::Result as StdResult;
+
+	fn repo() -> LocalGitRepo {
+		LocalGitRepo {
+			path: "/home/users/me/.cache/hipcheck/clones/github/expressjs/express/".to_string(),
```

Comment:
Add a comment that this is a dummy path that doesn't need to be real for these tests.